src/main/webapp/js/wikiSearch.js

```diff
@@ -0,0 +1,46 @@
+function searchOnWiki(){
+  fetch('/logo-detect').then(function(response) {
+    return response.json();
+  }).then((logos) => {
+    const query = logos[0];
+    //var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+    var xhr = new XMLHttpRequest();
+    const url = 'https://en.wikipedia.org/w/api.php?action=opensearch&search=' +
```

Comment:
Wikipedia's API has a feature, where if you provide `callback=someFunction` argument, the reply will be a `script` that calls `someFunction` with the result of the query. Try this to see what I mean: `https://en.wikipedia.org/w/api.php?action=opensearch&search=starbucks&limit=1&namespace=0&format=json&callback=handleSummary`

IIRC this format is called `jsonp`. The way it is commonly used is that you implement a callback function and then dynamically create a `<script>` element on the page with `src` set to the URL you want to fetch. Once the script is added to the page, it will be fetch and the callback that you have implemented will be called.

HTH

Reply:
See `JavaScript Example` on [this W3Schools page](https://www.w3schools.com/js/js_json_jsonp.asp)